src/Symfony/Component/Validator/Tests/Constraints/IsinValidatorTest.php

```diff
@@ -0,0 +1,221 @@
+<?php
+
+namespace Symfony\Component\Validator\Tests\Constraints;
+
+use Symfony\Component\Validator\Constraints\Isin;
+use Symfony\Component\Validator\Constraints\IsinValidator;
+use Symfony\Component\Validator\Test\ConstraintValidatorTestCase;
+
+class IsinValidatorTest extends ConstraintValidatorTestCase
+{
+    protected function createValidator()
+    {
+        return new IsinValidator();
+    }
+
+    public function testNullIsValid()
+    {
+        $this->validator->validate(null, new Isin());
+
+        $this->assertNoViolation();
+    }
+
+    public function testEmptyStringIsValid()
+    {
+        $this->validator->validate('', new Isin());
+
+        $this->assertNoViolation();
+    }
+
+    /**
+     * @dataProvider getValidIsin
+     */
+    public function testValidIsin($isin)
+    {
+        $this->validator->validate($isin, new Isin());
+        $this->assertNoViolation();
+    }
+
+    public function getValidIsin()
+    {
+        return [
+            ['XS2125535901'], 	// Goldman Sachs International	MTN	HKD HKD	126d
```

Comment:
These text fixtures doesn't look very useful ... they are all `X....`  What if we remove most of them and add instead some of the examples of the Isin Wikipedia page?

```php
['US0378331005'],   // Apple, Inc.
['AU0000XVGZA3'],   // TREASURY CORP VICTORIA 5 3/4% 2005-2016
['GB0002634946'],   // BAE Systems
```

Reply:
Done.